### DISCUSSION TITLE
support bram/fifo interface

### DISCUSSION THREAD

**Discussion by roybod** [esynr3z/corsair#127] [Questions]
we
re
addr
wdata
rdata

**Comment by esynr3z** [comment]
Hi!
FIFO interface is already supported and can be enabled for a field via `f` [hardware flag](https://corsair.readthedocs.io/en/latest/regmap.html#bit-field). You could also check result in [examples](https://github.com/esynr3z/corsair/blob/14dc5d40a52f7efac07393d0a53623fc231f8f77/examples/regmap_yaml/hw/regs.v#L12). However, interface is a bit different from suggested above - there's no address and FIFO can apply backpressure with write-ready and read-valid signals.

As for BRAM being mapped into register map - there is already issue for this: #39. This is planned for future versions.

**Comment by roybod** [comment]
thanks.
i came across your tool because i used AIRHDL and found out that the developer died...

**Comment by esynr3z** [comment]
I'm sorry to hear that. Is that tool now unmaintained?

Maybe, if you have some free minute, could you please share some feedback about airhdl? Maybe what was the best and worst parts of it? This could help Corsair in the future.

**Comment by roybod** [comment]
there is a website [airhdl.com](url)
it is still working
you can create a free user and test one register map
here is one example in zip file
[aurora_regs_rev47.zip](https://github.com/user-attachments/files/17533373/aurora_regs_rev47.zip)

feel free to contact me at my work email rbodner@parallelwireless.com
